### DISCUSSION TITLE
Make Area Type more diverse

### DISCUSSION THREAD

**Discussion by tane64** [jseidl/magic-areas#183] [Ideas / Feature Requests]
Testing the Magic Areas Looks very promising.

Few req:
I like to use the "Type" as section
Type is currently limited. Like to add moer section to Type. (First floor, second, Attic, front, garden etc.)

**Comment by jseidl** [comment]
Can you elaborate on what you mean by `Type` and `Section`?

**Comment by tane64** [comment]
The last select box (Area type (interior/exterior)
I like yo have more selections, First-flor, second-floor, Attic, garden, Front, Living. Other option is just same basic names Area1, Area2, Area3 etc.. T
Also option to select more then one.

Sample:
Like to turn off the lights in the living when there is no presence.
My living is Entrance, kitchen, dinning, TV room.

**Comment by jseidl** [comment]
Got it, so Area Type would actually be a multi-select (as area can be both interior and first-floor). Breaking change so it will have to wait for another major release. Thanks for the idea!

**Comment by icepod** [comment]
I think the expected behavior here would be similar to "Zones" in HomeKit.
There you can assign a "Room" ("Area") to multiple "Zones" ("Type"); and vice-versa.

This enables multiple user-defined criteria, for example:

Home > Indoor > First Floor > Private Rooms > Master > Bedroom
      and
Home > Outdoor > First Floor > Private Rooms > Master > Balcony
   or
Home > Indoor > Ground Floor > Private Rooms > Office
      and
Home > Indoor > Ground Floor > Public Room > Bathroom

This is useful for filtering and/or grouping by criteria that are either hierarchical or inclusive/exclusive in nature.

Maybe worth mentioning before the implementation is underway.

I could use this feature in Home Assistant as well!

**Comment by ReplicaObscura** [comment]
When you discuss Type, it sounds almost like simple categorization. But then based on your later description and icepod's explanation, it actually seems hierarchical, like some areas are parents of other areas.

I can see the usefulness of that hierarchy, being able to essentially roll all devices up to their parent area so you can control entities of the children as if they were in the parent.

So I fully agree with this, but I can't help but think that functionality would be more beneficial as a Home Assistant feature than as a Magic Areas feature. While it would be useful in MA, I think it would have many more potential use cases outside of MA in the rest of HA if areas could act hierarchically.

But, maybe doing it in MA would be a good step in that direction either way.

As an alternate way than simply defining more area types, perhaps each area in MA could define an optional parent area, and if there is no parent, then it rolls up to the main Interior or Exterior one depending on which is selected?

**Comment by jseidl** [comment]
The day has come! https://github.com/jseidl/hass-magic_areas/pull/342

FLoor meta areas coming next release

**Comment by jseidl** [comment]
This is implemented in release 4.0.0 which is out!
